### PR TITLE
Fix #230

### LINKED ISSUE #230: Request eliminating the location order limitation on the dependent job
**Is your feature request related to a problem? Please describe.**
When we add an Infrabox job A with dependency on job B, the dependency job B must be defined in front of the job A. Otherwise we will hit job parse error when pushing the Infrabox json file. Is it possible to eliminate the location order limitation?

**Describe the solution you'd like**
When we add an Infrabox job A with dependency on job B, we can define the job B before/after the job A which means the location of job B is insensitive.

### PR DESCRIPTION
**Pull request by iberryful**
#230

### CODE REVIEW

**Comment by iberryful** [comment]
I will create a PR for infrabox-cli later

File: src/pyinfrabox/infrabox/__init__.py [hunk 1]
```diff
@@ -537,7 +538,7 @@ def validate_json(d):
             if job_name == parent_name:
                 raise ValidationError(path, "Job '%s' may not depend on itself" % parent_name)
 
-            if parent_name not in jobs:
+            if parent_name not in all_job_names:
                 raise ValidationError(path + ".depends_on", "Job '%s' not found" % parent_name)
```
Comment (ib-steffen) [inline src/pyinfrabox/infrabox/__init__.py]: now we need an additional check that a job does not depend on itself.

Comment (iberryful) [reply src/pyinfrabox/infrabox/__init__.py]: This situation is checked in line 537

Comment (iberryful) [reply src/pyinfrabox/infrabox/__init__.py]: I think we should check  if there is a loop in dependency graph

Comment (ib-steffen) [reply src/pyinfrabox/infrabox/__init__.py]: yes it's already checked, but you are  right cyclic dependencies have to be checked

**Comment by ib-steffen** [comment]
the tests have to be adapted as well

**Pull request closed without merging**